doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,159 @@
+C++ Plugin API for CPU compressors  {#data}
+============
+
+This document specifies a modern API for the in-memory C plugin API for
+lossless byte data compressors. It will not replace the 2D image
+compressors and CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current lossless byte compressors
+    * Lossless compression of binary blobs
+    * Multiple compressors per plugin DSO
+* Information provided for each compressor:
+    * Compression speed (relative to RLE compressor)
+    * Compression ratio (relative to uncompressed data)
+* Compressors may produce multiple output chunks
+* Compressor engines are instantiated to allow reuse of internal state
+  and memory allocations
+* Easy to use API for plugin implementers
+
+## Dependency Changes
+
+* New: Using Lunchbox plugin API for loading plugins
+
+## API
+
+    const Plugins& Lunchbox::PluginFactory<..>::getPlugins();
```

Comment:
This method is still not present, right? Make it clear that is an addition.
You haven't specified what Plugins is. Do not omit the template parameters, use a template declaration.
I also have the feeling that this method is an implementation detail and it's not really part of the public API.

I guess you've already implemented this but how does it work with all the plugin factories we already have? If I take Brion::Morphology::Impl::MorphologyPluginFactory as an example, I see several issues with this method as is:
- It requires the derived classes of the template type T (assuming that Plugins is std::vector<T*> based on the example code) to be default constructible, and this is not the case for the morphology Plugins: they all use RAII and take MorphologyInitData as the constructor parameter.
- For morphologies this method is useless, because the base plugin class doesn't provide any useful information to distinguish the different plugins.

Or maybe I got it all wrong because Plugins doesn't contain plugin instances but the lunchbox::Plugin<T> stored by the factory.